### ISSUE TITLE
Error grouping multiple filepond fields using array notation

### ISSUE THREAD

**Issue by peterlcsteele** [rahulhaque/laravel-filepond#47]
Hi rahulhaque,

Thanks again for your work on this awesome project. I have an unusual scenario where I'm trying to group filepond upload files using array notation.

I'm experiencing Error 500 response when I try to upload a file to the second field.

I'm using version: 1.12.13.

Here is a sample of what I'm trying to achieve. You'll notice I'm using `name="assets[pptx]"` and `name="assets[pdf]"`

```html
<label for="pptx" class="wsm-label">PowerPoint</label>
<input id="pptx" class="filepond" type="file" name="assets[pptx]" required data-label-idle="Drop or select a PowerPoint file (.pptx)" data-accepted-file-types="application/vnd.openxmlformats-officedocument.presentationml.presentation" />
<p class="help-block">{{ $errors->first('assets[pptx]') }}</p>
                                        
<label for="pdf" class="wsm-label">PDF</label>
<input id="pdf" class="filepond" type="file" name="assets[pdf]" required data-label-idle="Drop or select a PDF file (.pdf)" data-accepted-file-types="application/pdf" />
<p class="help-block">{{ $errors->first('assets[pdf]') }}</p>
```

My JS config is as follows:
```js
// Register plugins
FilePond.registerPlugin(FilePondPluginFileValidateSize);
FilePond.registerPlugin(FilePondPluginFileValidateType);

// Set default FilePond options
FilePond.setOptions({
  // Enabled chunked uploads
  chunkUploads: true,
  server: {
    url: "{{ config('filepond.server.url') }}",
    headers: {
      'X-CSRF-TOKEN': "{{ @csrf_token() }}",
    }
  }
});

FilePond.parse(document.body);
```

Is array notation supported? If so, any support, suggestions and/or insights you can provide about why I'm receiving Error 500 would be much appreciated. If the fields are named uniquely without array notation: `name="assets_pptx"` `name="assets_pdf"`, I do not experience this issue and everything works perfectly.

**Comment by rahulhaque** [comment]
Hi @peterlcsteele, welcome here. I am not sure if this is a valid approach? I'll have to test first. BTW what's the error you're getting in response? What do you get in request after submitting the form? Most importantly how are you retrieving the file with my package?

**Comment by peterlcsteele** [comment]
Hey @rahulhaque,

If not a valid approach, that is fine, I can work around it by merging the assets like so:

```
$assets = collect($request->all())->reduce(function ($acc, $value, $key) {
  if (preg_match('/^assets_/', $key)) {
    $newKey = preg_replace('/^assets_/', '', $key);
    $acc[$newKey] = $value;
  }
  return $acc;
}, []);
$request->merge(['assets' => $assets]);
```

I haven't done much server-side diagnosis, but this is what I get on the client side.

```ErrorException: Undefined array key 0 in file .../vendor/rahulhaque/laravel-filepond/src/Services/FilepondService.php on line 38```

The following is then rendered for the second PDF filepond:
![array-syntax-ui-error](https://github.com/rahulhaque/laravel-filepond/assets/12838081/3279f686-adca-4d4a-8806-1690e673fd8d)

In terms of retrieving and handling the response server side. I had a working example using `name="assets_pptx"` and `name="assets_pdf"`, but I haven't got this far with the array syntax. Using array syntax, I expected to be able to write the following in the controller:

```
$this->validate($request, [
  'assets[pptx]' => Rule::filepond([
    // File ext: pptx
   'mimes:pptx',
    // Max size: 80mb
    'max:80000'
  ]),
  'assets[pdf]' => Rule::filepond([
    // File ext: pdf
    'mimes:pdf',
    // Max size: 80mb
    'max:80000'
  ])
]);
```

Following validation, I would move uploaded files from filepond temp directory to final destination and save the file path to the database.

```
$path = "/path/to/folder";
Filepond::field($request->assets['pptx'])->moveTo($path);
Filepond::field($request->assets['pdf'])->moveTo($path);
// TODO: Save destination paths to database.
```

This approach worked fine when not using array syntax.

**Comment by rahulhaque** [comment]
@peterlcsteele thanks for the details. I'll have a look. Is it that your first (PowerPoint) input works and the second (PDF) input doesn't in the same form?

**Comment by peterlcsteele** [comment]
Thanks @rahulhaque - Yep, correct. That has been my experience so far. Both filepond elements have the `required` attribute. I haven't yet experimented removing this attribute and then checking what the $request contains.

**Comment by rahulhaque** [comment]
@peterlcsteele I have uploaded some changes in the dev branch. Can you please test by installing with - `composer require rahulhaque/laravel-filepond:dev-dev`. If successful, try retrieving the file with - `Filepond::field($request->input('assets.pptx'))->getFile()`.

**Comment by peterlcsteele** [comment]
Good job @rahulhaque. 

I can confirm that installing the dev branch fixes the issue.

Here's a dd() of the $request:
<img width="1291" alt="dd-request" src="https://github.com/rahulhaque/laravel-filepond/assets/12838081/196c095b-9c2f-42a4-ac71-0137ed0f1c2e">

And heres a dd() after getting both models:
<img width="1526" alt="dd-models" src="https://github.com/rahulhaque/laravel-filepond/assets/12838081/a9f0c68b-db1b-4a34-a927-15d5661f16d7">

I appreciate you investigating and resolving this so quickly.

**Comment by rahulhaque** [comment]
@peterlcsteele please, update to the new release and close the issue when done testing.

**Comment by peterlcsteele** [comment]
Resolved in v1.12.14

Thanks again to @rahulhaque for this awesome package.
